File: src/main/java/fgo/cards/fgo/BlessingOfKur.java
package fgo.cards.fgo;

import com.megacrit.cardcrawl.actions.common.ApplyPowerAction;
import com.megacrit.cardcrawl.characters.AbstractPlayer;
import com.megacrit.cardcrawl.monsters.AbstractMonster;

import fgo.cards.FGOCard;
import fgo.cards.noblecards.KurKigalIrkalla;
import fgo.powers.BlessingOfKurPower;
import fgo.powers.NPRatePower;

public class BlessingOfKur extends FGOCard {
    public static final String ID = makeID(BlessingOfKur.class.getSimpleName());

    public BlessingOfKur() {
        super(ID, 1, CardType.POWER, CardTarget.SELF, CardRarity.UNCOMMON);
        setMagic(6, 9);
        cardsToPreview = new KurKigalIrkalla();
    }

    @Override
    public void use(AbstractPlayer p, AbstractMonster m) {
        addToBot(new ApplyPowerAction(p, p, new NPRatePower(p, 3)));
        addToBot(new ApplyPowerAction(p, p, new BlessingOfKurPower(p, magicNumber, 2)));
    }
}

File: Scripts/Cards/BlessingOfKur.cs
using Fgo.Scripts.Cards.NoblePhantasm;
using Fgo.Scripts.Powers;
using MegaCrit.Sts2.Core.Commands;
using MegaCrit.Sts2.Core.Entities.Cards;
using MegaCrit.Sts2.Core.GameActions.Multiplayer;
using MegaCrit.Sts2.Core.HoverTips;
using MegaCrit.Sts2.Core.Localization.DynamicVars;
using STS2RitsuLib.Cards.DynamicVars;

namespace Fgo.Scripts.Cards;

public class BlessingOfKur() : FgoCardModel(1, CardType.Power,
    CardRarity.Uncommon, TargetType.Self)
{
    private const int BaseStacks = 2;
    private const int UpgradeStacks = 1;

    protected override IEnumerable<IHoverTip> AdditionalHoverTips =>
    [
        HoverTipFactory.FromCard<KurKigalIrkalla>(),
        HoverTipFactory.FromPower<BlessingOfKurPower>()
    ];

    protected override IEnumerable<DynamicVar> CanonicalVars =>
    [
        ModCardVars.Power<BlessingOfKurPower>(BaseStacks),
        ModCardVars.Power<MaxHpPower>(BaseStacks * BlessingOfKurPower.MaxHpPerStack)
    ];

    protected override void OnUpgrade()
    {
        DynamicVars[nameof(BlessingOfKurPower)].UpgradeValueBy(UpgradeStacks);
        DynamicVars[nameof(MaxHpPower)].UpgradeValueBy(UpgradeStacks * BlessingOfKurPower.MaxHpPerStack);
    }

    protected override async Task OnPlay(PlayerChoiceContext choiceContext, CardPlay cardPlay)
    {
        await PowerCmd.Apply<NpRatePower>(choiceContext, Owner.Creature, 3m, Owner.Creature, this);
        await PowerCmd.Apply<BlessingOfKurPower>(choiceContext, Owner.Creature,
            DynamicVars[nameof(BlessingOfKurPower)].BaseValue,
            Owner.Creature, this);
    }
}